--- a/net/minecraft/src/ChunkProviderHell.java
+++ b/net/minecraft/src/ChunkProviderHell.java
@@ -35,7 +35,7 @@
 		this.field_4176_b = new NoiseGeneratorOctaves(this.hellRNG, 16);
 	}
 
-	public void func_4059_a(int i1, int i2, byte[] b3) {
+	public void func_4059_a(int i1, int i2, short[] b3) {
 		byte b4 = 4;
 		byte b5 = 32;
 		int i6 = b4 + 1;
@@ -100,7 +100,7 @@
 
 	}
 
-	public void func_4058_b(int i1, int i2, byte[] b3) {
+	public void func_4058_b(int i1, int i2, short[] b3) {
 		byte b4 = 64;
 		double d5 = 8.0D / 256D;
 		this.field_4162_p = this.field_4166_l.generateNoiseOctaves(this.field_4162_p, (double)(i1 * 16), (double)(i2 * 16), 0.0D, 16, 16, 1, d5, d5, 1.0D);
@@ -123,7 +123,7 @@
 					} else if(i15 <= 0 + this.hellRNG.nextInt(5)) {
 						b3[i16] = (byte)Block.bedrock.blockID;
 					} else {
-						byte b17 = b3[i16];
+						short b17 = b3[i16];
 						if(b17 == 0) {
 							i12 = -1;
 						} else if(b17 == Block.netherrack.blockID) {
@@ -179,7 +179,7 @@
 
 	public Chunk provideChunk(int i1, int i2) {
 		this.hellRNG.setSeed((long)i1 * 341873128712L + (long)i2 * 132897987541L);
-		byte[] b3 = new byte[32768];
+		short[] b3 = new short[32768];
 		this.func_4059_a(i1, i2, b3);
 		this.func_4058_b(i1, i2, b3);
 		this.field_4159_s.func_867_a(this, this.worldObj, i1, i2, b3);
