--- a/net/minecraft/src/ChunkProviderClient.java
+++ b/net/minecraft/src/ChunkProviderClient.java
@@ -13,7 +13,7 @@
 	private World worldObj;
 
 	public ChunkProviderClient(World world1) {
-		this.blankChunk = new EmptyChunk(world1, new byte[32768], 0, 0);
+		this.blankChunk = new EmptyChunk(world1, new short[32768], 0, 0);
 		this.worldObj = world1;
 	}
 
@@ -38,7 +38,7 @@
 
 	public Chunk prepareChunk(int i1, int i2) {
 		ChunkCoordIntPair chunkCoordIntPair3 = new ChunkCoordIntPair(i1, i2);
-		byte[] b4 = new byte[32768];
+		short[] b4 = new short[32768];
 		Chunk chunk5 = new Chunk(this.worldObj, b4, i1, i2);
 		Arrays.fill(chunk5.skylightMap.data, (byte)-1);
 		this.chunkMapping.put(chunkCoordIntPair3, chunk5);
