--- /dev/null
+++ b/net/minecraft/src/EnumToolMaterialLonsdalite.java
@@ -1,0 +1,31 @@
+package net.minecraft.src;
+
+public enum EnumToolMaterialLonsdalite {
+    MATERIALLONSDALITE(3, 3122, 16.0F, 0, 6);
+    private final int harvestLevel;
+    private final int maxUses;
+    private final float efficiencyOnProperMaterial;
+    private final int damageVsEntity;
+    EnumToolMaterialLonsdalite(int par3, int par4, float par5, int par6, int i) {
+        harvestLevel = par3;
+        maxUses = par4;
+        efficiencyOnProperMaterial = par5;
+        damageVsEntity = par6;
+    }
+
+    public int getMaxUses() {
+        return maxUses;
+    }
+
+    public float getEfficiencyOnProperMaterial() {
+        return efficiencyOnProperMaterial;
+    }
+
+    public int getDamageVsEntity() {
+        return damageVsEntity;
+    }
+
+    public int getHarvestLevel() {
+        return harvestLevel;
+    }
+}
