--- a/net/minecraft/src/NBTBase.java
+++ b/net/minecraft/src/NBTBase.java
@@ -66,6 +66,8 @@
 			return new NBTTagList();
 		case 10:
 			return new NBTTagCompound();
+		case 11:
+			return new NBTTagShortArray();
 		default:
 			return null;
 		}
@@ -95,6 +97,8 @@
 			return "TAG_List";
 		case 10:
 			return "TAG_Compound";
+		case 11:
+			return "TAG_ShortArray";
 		default:
 			return "UNKNOWN";
 		}
